test/src/specs/relay/compact_block.rs

```diff
@@ -275,32 +329,25 @@ impl CompactBlockBasic {
         let old_tip = node.get_tip_block().header().number();
 
         net.send(
-            NetworkProtocol::SYNC.into(),
+            NetworkProtocol::RELAY.into(),
             peer_id,
-            build_header(parent.header()),
+            build_compact_block(&parent),
         );
+        // pending for GetBlockTransactions
+        clear_messages(&net);
+
         net.send(
-            NetworkProtocol::RELAY.into(),
+            NetworkProtocol::SYNC.into(),
             peer_id,
-            build_compact_block(&parent),
+            build_header(parent.header()),
         );
         net.send(
             NetworkProtocol::SYNC.into(),
             peer_id,
             build_header(block.header()),
         );
+        clear_messages(&net);
 
-        // Wait until node0 send GetBlocks
-        loop {
-            let (_, _, data) = net
-                .receive_timeout(Duration::from_secs(5))
-                .expect("wait GetBlocks");
-            if let Ok(message) = get_root::<SyncMessage>(&data) {
-                if message.payload_type() == SyncPayload::GetBlocks {
-                    break;
-                }
-            }
-        }
```

Comment:
Without this loop-block, I am afraid that the test would be failed occasionally.
For example, `node0` has received `Headers[parent, block]` but not sends `GetBlocks` yet, which means `blocks_inflight` doesn't contain `[parent, block]`. So if you send `Blocks[block]` to `node0`, `node0` will ignore it since [here](https://github.com/nervosnetwork/ckb/blob/6f39901c1a04d62fd55b6131a23e62e32029055b/sync/src/synchronizer/block_process.rs#L41)

Reply:
> Without this loop-block, I am afraid that the test would be failed occasionally.
> For example, `node0` has received `Headers[parent, block]` but not sends `GetBlocks` yet, which means `blocks_inflight` doesn't contain `[parent, block]`. So if you send `Blocks[block]` to `node0`, `node0` will ignore it since [here](https://github.com/nervosnetwork/ckb/blob/6f39901c1a04d62fd55b6131a23e62e32029055b/sync/src/synchronizer/block_process.rs#L41)

I have changed this loop to `clear_messages` to make sure that `GetBlocks` has been received:
https://github.com/nervosnetwork/ckb/pull/1141/files#diff-be3de08a9c2e92180bffa68b2ffb89b1R349

Reply:
Oh...you increase the timeout duration of `clear_messages`...